extensions/hazelcast-client/runtime/src/main/java/io/quarkus/hazelcast/client/runtime/HazelcastConfigurationParser.java

```diff
@@ -0,0 +1,75 @@
+package io.quarkus.hazelcast.client.runtime;
+
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import com.hazelcast.client.config.ClientConfig;
+
+class HazelcastConfigurationParser {
+
+    ClientConfig fromApplicationProperties(HazelcastClientConfig config, ClientConfig clientConfig) {
+        setClusterAddress(clientConfig, config);
+        setGroupName(clientConfig, config);
+        setLabels(clientConfig, config);
+
+        setOutboundPorts(clientConfig, config);
+        setOutboundPortDefinitions(clientConfig, config);
+
+        setConnectionTimeout(clientConfig, config);
+        setConnectionAttemptLimit(clientConfig, config);
+        setConnectionAttemptPeriod(clientConfig, config);
+
+        setExecutorPoolSize(clientConfig, config);
+
+        return clientConfig;
+    }
+
+    private void setClusterAddress(ClientConfig clientConfig, HazelcastClientConfig config) {
+        config.clusterMembers
+                .ifPresent(members -> clientConfig.getNetworkConfig().addAddress(members.split(",")));
```

Comment:
Your config should be a List and the split on `,` will be done automatically by the config framework.